crates/lib3h/src/transport/transport_multiplex/route.rs

```diff
@@ -0,0 +1,235 @@
+            RequestToParentResponse,
+            TransportError,
+        >,
+    >,
+}
+
+impl TransportMultiplexRoute {
+    /// create a new TransportMultiplexRoute Instance
+    pub(crate) fn new(
+        route_spec: LocalRouteSpec,
+        inner_transport: GhostEndpoint<
+            RequestToChild,
+            RequestToChildResponse,
+            RequestToParent,
+            RequestToParentResponse,
+            TransportError,
+        >,
+    ) -> Self {
+        let (endpoint_parent, endpoint_self) = create_ghost_channel();
+        let endpoint_parent = Some(endpoint_parent);
+        let endpoint_self = Detach::new(endpoint_self.as_context_endpoint("route_to_parent_"));
+        let inner_transport = Detach::new(inner_transport.as_context_endpoint("route_to_inner_"));
+        Self {
+            route_spec,
+            endpoint_parent,
+            endpoint_self,
+            inner_transport,
+        }
+    }
+
+    /// private dispatcher for messages from our inner transport
+    fn handle_msg_from_inner(
+        &mut self,
+        mut msg: GhostMessage<
+            RequestToParent,
+            RequestToChild,
+            RequestToParentResponse,
+            TransportError,
+        >,
+    ) -> TransportResult<()> {
+        match msg.take_message().expect("exists") {
+            RequestToParent::IncomingConnection { address } => {
+                self.handle_incoming_connection(address)
+            }
+            RequestToParent::ReceivedData { address, payload } => {
+                self.handle_received_data(address, payload)
+            }
+            RequestToParent::TransportError { error } => self.handle_transport_error(error),
+        }
+    }
+
+    /// private handler for inner transport IncomingConnection events
+    fn handle_incoming_connection(&mut self, address: Url) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::IncomingConnection { address });
+        Ok(())
+    }
+
+    /// private handler for inner transport ReceivedData events
+    fn handle_received_data(&mut self, address: Url, payload: Vec<u8>) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::ReceivedData { address, payload });
+        Ok(())
+    }
+
+    /// private handler for inner transport TransportError events
+    fn handle_transport_error(&mut self, error: TransportError) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::TransportError { error });
+        Ok(())
+    }
+
+    /// private dispatcher for messages coming from our parent
+    fn handle_msg_from_parent(
+        &mut self,
+        mut msg: GhostMessage<
+            RequestToChild,
+            RequestToParent,
+            RequestToChildResponse,
+            TransportError,
+        >,
+    ) -> TransportResult<()> {
+        match msg.take_message().expect("exists") {
```

Comment:
Same comment- Why is message optional here and assumed to exist? I'll stop commenting on these- think you get the idea.